packages/shared/src/components/drawers/ListDrawer.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { ReactElement } from 'react';
+import classNames from 'classnames';
+import { Drawer, DrawerRef, DrawerWrapperProps } from './Drawer';
+import { VIcon } from '../icons';
+import { IconSize } from '../Icon';
+import { SelectParams } from './common';
+
+interface ListDrawerProps {
+  drawerProps: Omit<DrawerWrapperProps, 'children'>;
+  options: string[];
+  selected: number; // index
+  onSelectedChange(props: SelectParams): void;
+}
+
+export function ListDrawer({
+  onSelectedChange,
+  drawerProps,
+  selected,
+  options,
+}: ListDrawerProps): ReactElement {
+  const ref = React.useRef<DrawerRef>();
+
+  return (
+    <Drawer {...drawerProps} ref={ref} role="menu">
+      {options.map((value, index) => {
+        const isSelected = index === selected;
+
+        return (
```

Comment:
Isn't this basically the same as ContextMenuDrawer?
Or atleast many common parts right?

Maybe we can fully abstract, but atleast abstract to common classed component?

Reply:
For the ContextMenuDrawer, I think the primary difference is the abstracted check icon for selected items.

Reply:
Hmm, wondering if it's worth combining them?
It will just make our lives easier down the line maybe.. (but open to both to be honest)

Reply:
Yeah, the context menu components feel like a bit all over the place since we implemented a generic one but then never checked old implementations to be migrated.